### ISSUE TITLE
Thoughts after testing this method.

### ISSUE THREAD

**Issue by random-yang** [andrewyng/translation-agent#10]
Is this all expected?

1. Even if using gpt3.5 in combination with gpt4, the speed is still extremely slow
2. There is still the characteristic of a large model - instability
3. There is an effect of improvement, but occasionally there will also be negative optimization
4. And the token cost is several times that of a single translation

**Comment by siddhantx0** [comment]
Yeah I think so.

**Comment by hhhaiai** [comment]
It's too expensive

**Comment by random-yang** [comment]
> It's too expensive

Yes, so I packaged it into a coze bot, and it is still free, so other than being a little slow, it’s pretty good!
If you guys interested, here is the link: https://www.coze.com/s/Zs8MEgGcR/

**Comment by j-dominguez9** [comment]
Hey @random-yang , thanks for your issue. Yes, I think all those points that you mentioned are expected. For clarity, however, I figured I could elaborate on them somewhat:
1. We were only focused on optimizing for quality of output in this process, not latency; although, I recognize how that could be important to people/some applications.
2. The non-determinism of a model isn't something we can control, but we can see to what extent it affects the application by evaluating the outputs and tweaking the process accordingly.
3. Good point! It could be worth thinking about checking to what degree translations are made "worse" by reflection, which we have found tends to be a somewhat subjective term in MT.
4. That is true. As mentioned, we were trying to optimize for translation quality. We hope this process can be applied with local open-source (free) models as well.

Closing this issue for now, due to no resolution, but if you have suggestions about how to address these points, we gladly welcome a proposal!
